### PR TITLE
Service name

### PR DESCRIPTION
**Pull request by notkriswagner**
Added support for service name as well as completed a refactoring of request headers. A mock test framework was also added to verify that service name works.

### CODE REVIEW

**Comment by notkriswagner** [comment]
@smtakeda please review when you get a chance

File: lib/client_int.h [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #define HEADER_ACCEPT_TYPE_APPLICATION_JSON "accept: application/json"
 #define HEADER_C_API_USER_AGENT "User-Agent: c_api/0.1"
 #define HEADER_DIRECT_QUERY_TOKEN_FORMAT "Authorization: %s"
+#define HEADER_SERVICE_NAME_FORMAT "Service-Name: %s"
```
Comment (smtakeda) [inline lib/client_int.h]:
```
X-Snowflake-Service
```
?

**Comment by smtakeda** [comment]
LGTM other than the header name.

File: lib/client_int.h (hunk 1 above)
Comment (notkriswagner) [reply lib/client_int.h]: Good catch

**Pull request merged**